wgpu-core/src/command/bundle.rs

```diff
@@ -667,13 +669,45 @@ pub enum RenderBundleError {
     Draw(#[from] DrawError),
 }
 
+type RenderBundleCommandError = Option<&'static str>;
+
+/// Error encountered when finishing recording a render bundle.
+#[derive(Clone, Debug, Error)]
+#[error("{command}")]
+pub struct RenderBundleError {
+    command: &'static str,
+    #[source]
+    inner: RenderBundleErrorInner,
+}
+
+impl RenderBundleError {
+    fn new(command: RenderBundleCommandError, inner: RenderBundleErrorInner) -> Self {
+        RenderBundleError {
+            command: command.unwrap_or("In the bundle parameters"),
+            inner,
+        }
+    }
+}
+
 impl<G: GlobalIdentityHandlerFactory> Global<G> {
     pub fn render_bundle_encoder_finish<B: GfxBackend>(
         &self,
         bundle_encoder: RenderBundleEncoder,
         desc: &RenderBundleDescriptor,
         id_in: Input<G, id::RenderBundleId>,
     ) -> Result<id::RenderBundleId, RenderBundleError> {
+        let mut command: RenderBundleCommandError = None;
+        self.render_bundle_encoder_finish_inner::<B>(bundle_encoder, desc, id_in, &mut command)
+            .map_err(|inner| RenderBundleError::new(command, inner))
+    }
+
+    fn render_bundle_encoder_finish_inner<B: GfxBackend>(
+        &self,
+        bundle_encoder: RenderBundleEncoder,
+        desc: &RenderBundleDescriptor,
+        id_in: Input<G, id::RenderBundleId>,
+        err_ctx: &mut RenderBundleCommandError,
```

Comment:
Can we get away from this mutable semantic entirely? Generally we should try to keep this "functional style". Mixing the procedural and function in the same semantics (error passing) is awkward, and would impede maintenance of this code.

Can we instead just have `-> Result<id::RenderBundleId, RenderBundleError>`? whenever there is a spot that returns an error, we'd just map that error and insert the appropriate command description in place, i.e.

```rust
something()
  .map_err(|error| RenderBundleError {
    command: "me does things wrong",
    inner: error.into(),
  })?

Reply:
I did something like that initially, but it got very verbose and error prone in my opinion, which is why I went with this blanked approach. 

I can try changing the mutable context setting to a more declarative style, but it probably would make a bigger diff as I need to capture all subcommands in a closure.

Reply:
Could you show how it looks for one small thing, e.g. compute errors? Just to get the idea